### PR TITLE
smac_planner: fix goal checking in getAnalyticPath()

### PR DESCRIPTION
**Pull request by mkolodziejczyk-piap**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #2040 |
| Primary OS tested on | Ubuntu |
| Robotic platform tested on | my own simulation stack |

---

## Description of contribution in a few bullet points

* Added condition to exclude goal from poses added through analytic path expansion, because the goal is added always at the end and adding it twice causes infinite loops in the path

### CODE REVIEW

File: smac_planner/src/a_star.cpp [hunk 1]
```diff
@@ -411,7 +411,7 @@ AStarAlgorithm<NodeSE2>::NodePtr AStarAlgorithm<NodeSE2>::getAnalyticPath(
   prev = node;
   for (const auto & node_pose : possible_nodes) {
     const auto & n = node_pose.first;
-    if (!n->wasVisited()) {
+    if (!n->wasVisited() and n->getIndex()!=_goal->getIndex()) {
```
Comment (SteveMacenski) [inline smac_planner/src/a_star.cpp:414]:
```suggestion
    if (!n->wasVisited() && n->getIndex() != _goal->getIndex()) {
```

Just to be consistent in styling

**Review by SteveMacenski — CHANGES_REQUESTED** [review]
Just a styling change but LGTM. Can you explain in a few sentences what the specific issue you hit? I'm curious how we didn't also run into this in our testing.

**Comment by mkolodziejczyk-piap** [comment]
IMO the issue arises, when the vehicle is crossing the goal but the orientation is not the goal orientation and after crossing this goal ompl plans to correct the yaw with some back and forth manouvers. In my case, this loop happend almost all the time near the goal, probably because (1) my controller was not tuned enought to follow the path exactly (so the initial "smooth" global plan turned into some wild manouvers to correct the yaw) (2) my vehicle has large turning radius

**Comment by SteveMacenski** [comment]
> vehicle is crossing the goal but the orientation is not the goal orientation

Ooooooh, yeah I did _not_ test for that. Thanks!

Please make the update / `main` PR and we can merge the set.

**Comment by SteveMacenski** [comment]
@mkolodziejczyk-piap just following up here. Just need the small styling update & `main` PR to merge both

**Comment by SteveMacenski** [comment]
@mkolodziejczyk-piap just following up

This is essentially done, just needs superficial update

**Comment by SteveMacenski** [comment]
Closing - superseded by linked PRs

**Pull request closed without merging**